src/main/java/uk/gov/hmcts/ccd/domain/service/createevent/DefaultCreateEventOperation.java

```diff
@@ -117,33 +121,16 @@ public CaseDetails createCaseEvent(final String caseReference,
                                        final CaseDataContent content) {
         eventValidator.validate(content.getEvent());
 
-        final CaseDetails caseDetails = lockCaseDetails(caseReference);
-        final CaseType caseType = caseDefinitionRepository.getCaseType(caseDetails.getCaseTypeId());
-        final CaseEvent eventTrigger = findAndValidateCaseEvent(caseType, content.getEvent());
-        final CaseDetails caseDetailsBefore = caseService.clone(caseDetails);
-        String uid = userAuthorisation.getUserId();
+        CaseEventCreator caseEventCreator = transactionHelper.withNewTransaction(
+            () -> new CaseEventCreator().create(caseReference, content));
 
-        eventTokenService.validateToken(content.getToken(), uid, caseDetails, eventTrigger, caseType.getJurisdiction(), caseType);
+        CaseDetails caseDetails = caseEventCreator.getCaseDetails();
 
-        validatePreState(caseDetails, eventTrigger);
-        mergeUpdatedFieldsToCaseDetails(content.getData(), caseDetails, eventTrigger, caseType);
-        AboutToSubmitCallbackResponse aboutToSubmitCallbackResponse = callbackInvoker.invokeAboutToSubmitCallback(eventTrigger,
-            caseDetailsBefore,
-            caseDetails,
-            caseType,
-            content.getIgnoreWarning());
-
-        final Optional<String>
-            newState = aboutToSubmitCallbackResponse.getState();
-
-        validateCaseFieldsOperation.validateData(caseDetails.getData(), caseType);
-        final CaseDetails savedCaseDetails = saveCaseDetails(caseDetails, eventTrigger, newState);
-        saveAuditEventForCaseDetails(aboutToSubmitCallbackResponse, content.getEvent(), eventTrigger, savedCaseDetails, caseType);
-
-        if (!isBlank(eventTrigger.getCallBackURLSubmittedEvent())) {
+        if (!isBlank(caseEventCreator.getEventTrigger().getCallBackURLSubmittedEvent())) {
             try { // make a call back
                 final ResponseEntity<AfterSubmitCallbackResponse> callBackResponse = callbackInvoker
-                    .invokeSubmittedCallback(eventTrigger, caseDetailsBefore, savedCaseDetails);
+                    .invokeSubmittedCallback(caseEventCreator.getEventTrigger(), caseEventCreator.getCaseDetailsBefore(),
```

Comment:
rather than starting a nested transaction (increasing complexity perhaps?) to handle the save, perhaps we can reduce the scope of the current transaction to include only the save. I don't think we need the submitted callback to be inside the transaction